### DISCUSSION TITLE
Feature request: new edge events -> onEdgeSelected, onEdgeDeselected

### DISCUSSION THREAD

**Discussion by Zambiorix** [bcakmakoglu/vue-flow#1657] [Ideas]
Please consider a new feature:

Edge events: 
onEdgeSelected
onEdgeDeselected

Also, I have noticed that custom edge labels do not get a `selected` class assigned when the edge is selected.
At this moment I see no way to update an edge label based on selection.

Thanks
Cheers

Gerd

**Comment by bcakmakoglu** [comment]
You already get a flag called selected as a prop passed to custom edges. You can use that flag to toggle whatever you want on select/unselect without the need for additional events whatsoever

**Comment by Zambiorix** [comment]
Thanks! 

On a related note:
In my `custom` edge I use the following css to change the path color when selected.
But that does not change the colors of the arrow markers .. how do I do that?

```
.vue-flow__edge-custom path {
    stroke: #c0c0c0;
    stroke-width:2px
}

.vue-flow__edge-custom.selected path {
    stroke: #c36466 !important;
}
```

Thanks!

**Reply by bcakmakoglu** [reply]
Edge markers can be an object, which allow a property called `color` for example:

```ts
const edge = {
	id: 'e1->2',
	source: '1',
	target: '2',
	markerEnd: {
		type: MarkerType.ArrowClosed,
		color: '#000'
	}
}
```

If you want to change the marker color when an edge is selected you would have to update this object again with the respective color.
